tests/python/pants_test/goal/test_run_tracker.py

```diff
@@ -24,13 +24,19 @@ def test_upload_stats(self):
     class Handler(http.server.BaseHTTPRequestHandler):
       def do_POST(handler):
         try:
-          self.assertEqual('/upload', handler.path)
-          self.assertEqual('application/x-www-form-urlencoded', handler.headers['Content-type'])
-          length = int(handler.headers['Content-Length'])
-          post_data = parse_qs(handler.rfile.read(length).decode('utf-8'))
-          decoded_post_data = {k: json.loads(v[0]) for k, v in post_data.items()}
-          self.assertEqual(stats, decoded_post_data)
-          handler.send_response(200)
+          if handler.path.startswith('/redirect'):
+            code = int(handler.path[-3:])
+            handler.send_response(code)
+            handler.send_header('location', mk_url('/upload'))
```

Comment:
Wow... python scoping is strange.

Reply:
```
try:
    handler = html.lines()
except Exception:
    raise("x")
handler.stuff()
```
I try to defend python, but it is tough to justify leaked state like this...